addons/binding/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,241 @@
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.etherrain.EtherRainBindingConstants;
+import org.openhab.binding.etherrain.internal.EtherRainConfiguration;
+import org.openhab.binding.etherrain.internal.api.EtherRainCommunication;
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EtherRainHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+public class EtherRainHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EtherRainHandler.class);
+
+    private EtherRainCommunication device = null;
+    private boolean connected = false;
+
+    @Nullable
+    private ScheduledFuture<?> updateJob;
+
+    /*
+     * Constructor class. Only call the parent constructor
+     */
+    public EtherRainHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command.toString().equals("REFRESH")) {
+            updateBridge();
+            return;
+        }
+
+        else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_EXECUTE)) {
+            execute();
+            updateState(EtherRainBindingConstants.CHANNEL_ID_EXECUTE, OnOffType.OFF);
+        }
+
+        else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_CLEAR)) {
+            clear();
+            updateState(EtherRainBindingConstants.CHANNEL_ID_CLEAR, OnOffType.OFF);
+        } else {
+            logger.warn("Unhandled command: " + command.toFullString());
+        }
+
+    }
+
+    private boolean connectBridge() {
+        EtherRainConfiguration config = getConfigAs(EtherRainConfiguration.class);
+
+        logger.debug(
+                "Attempting to connect to Etherrain wtih config = (Hostname: {}, Port: {}, Password: {}, Refresh: {}).",
+                config.hostname, config.port, config.password, config.refresh);
+
+        device = new EtherRainCommunication(config.hostname, config.port, config.password);
+
+        try {
+            EtherRainStatusResponse response = device.commandStatus();
+            if (response == null) {
+                logger.debug("Command Status returned null");
+                device = null;
+                updateStatus(ThingStatus.OFFLINE);
+                return false;
+            }
+        } catch (Exception e) {
```

Comment:
Do you really need to catch `Exception` here? It is usally better to be more specific.